### PR TITLE
[Merged by Bors] - feat: generalize kernel Radon-Nikodym derivative to `CountableOrCountablyGenerated`

### PR DESCRIPTION
**Pull request by RemyDegenne**
Radon-Nikodym derivatives of finite kernels `kernel α γ` are currently defined under the assumption that `γ` is countably generated. We extend the construction to the case that either `α` is countable or `γ` is countably generated.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Probability/Kernel/RadonNikodym.lean [hunk 1]
```diff
@@ -9,7 +9,8 @@ import Mathlib.Probability.Kernel.WithDensity
 /-!
 # Radon-Nikodym derivative and Lebesgue decomposition for kernels
 
-Let `γ` be a countably generated measurable space and `κ, η : kernel α γ` be finite kernels.
+Let `α` and `γ` be two measurable space, where either `α` is countable or `γ` is countably genrated.
```
Comment (sgouezel) [inline Mathlib/Probability/Kernel/RadonNikodym.lean:12]:
```suggestion
Let `α` and `γ` be two measurable space, where either `α` is countable or `γ` is countably generated.
```

**Review by sgouezel — COMMENTED** [review]
bors d+
Thanks!

File: Mathlib/Probability/Kernel/RadonNikodym.lean [hunk 2]
```diff
@@ -9,7 +9,8 @@ import Mathlib.Probability.Kernel.WithDensity
 /-!
 # Radon-Nikodym derivative and Lebesgue decomposition for kernels
 
-Let `γ` be a countably generated measurable space and `κ, η : kernel α γ` be finite kernels.
+Let `α` and `γ` be two measurable space, where either `α` is countable or `γ` is countably generated.
+Let `κ, η : kernel α γ` be finite kernels.
```
Comment (RemyDegenne) [inline Mathlib/Probability/Kernel/RadonNikodym.lean:13]:
```suggestion
Let `α` and `γ` be two measurable space, where either `α` is countable or `γ` is
countably generated. Let `κ, η : kernel α γ` be finite kernels.
```

**Pull request closed without merging**